src/internal/middleware/errors/interceptor.go

```diff
@@ -22,12 +23,25 @@ func StreamServerInterceptor(srv interface{}, stream grpc.ServerStream, info *gr
 }
 
 func errorForGRPC(err error) error {
-	if _, ok := status.FromError(err); ok {
+	// strip off the initial stack trace layer, in case this is just a once-wrapped auth error
+	cursor := err
+	for cursor != nil {
+		if _, ok := cursor.(errors.StackTracer); ok {
+			cursor = errors.Unwrap(cursor)
+		} else {
+			break
+		}
+	}
+	if cursor == nil && err != nil {
+		// this should never happen, we unwrapped an error consisting only of stack traces
+		// reset to the original error, we have no idea what this is
+		cursor = err
+	}
+	if _, ok := status.FromError(cursor); ok {
 		// handles nil as well
-		return err //nolint:wrapcheck
+		return cursor //nolint:wrapcheck
```

Comment:
Any reason not to return `&gRPCStatusError{code: code, err: err}` from this branch?  That way we'd be keeping the stack trace, I think.  (Maybe gRPCStatusError should implement StackTracer and delegate to the underlying?  That would definitely complicate this PR, though, so maybe not worth it.)

Reply:
Yes, sorry, I should have been clearer. I explicitly want to strip the stack (the grpc handler code flattens the error immediately after, anyway) to avoid the double-wrapping issue that was happening with auth errors. The only issue was they were being wrapped with stack traces - I think it's fine for us to use `status.Error` in cases like that where the request hasn't even really started yet.

Reply:
Makes sense!